File: leetcode/march2024/python/29_count_subarrarys_where_max_element_comes_atleast_k_times.py
# CODE

class Solution:
    def countSubarrays(self, nums: List[int], k: int) -> int:
        maxiboi = max(nums)
        n = len(nums)
        res = start = end = 0
        for end in range(n):
            if nums[end] == maxiboi:
                k -= 1
            while k == 0:
                if nums[start] == maxiboi:
                    k += 1
                start += 1
            res += start
        return res

File: leetcode/march2024/c++/29_count_subarrarys_where_max_element_comes_atleast_k_times.cpp
// CODE

class Solution {
    public:
    long long countSubarrays(vector<int>& nums, int k) {
        int maxiboi = *max_element(nums.begin(), nums.end());
        int n = nums.size(), start = 0, end = 0;
        long long res = 0;
        while(end < n) {
            if(nums[end] == maxiboi) k--;
            while(k == 0) {
                if(nums[start] == maxiboi) k++;
                start++;
            }
            res += start;
            end++;
        }
        return res;
    }
};